Benny 40 crud operations members

bennytrieu opened the pull request:
- Made member's role less case sensitive
- Added an update, delete, and add member button option to only people that are signed in
- Implemented add, update, and delete member logic
- brute-force tested update, delete, and add member logic and made sure the functions work

The only bug I found after working on this is when a member is added/updated in the officers section, the roles are not sorted correctly for some reason. It might be because the dropdown options are capitalized, it doesn't know what number it is in but I could be wrong. Left some members for you to test delete/update/add members. DM me if there's any other potential odd bugs or if you want the list of how I tested the functions.

shortxmas commented:
nice guys this is so kewll

LeviKuhaulua reviewed (changes requested):
Add & Update Functionality: 

Member roles are being capitalized when added / updated. When can, please adjust so that roles are lowercase before being stored. 

CRUD Buttons: 

If these could be converted to icons instead. Specifically for the 'add' icon, this could be something that is added after the last member of that specific section. For example, after the last club advisor, there could be a plus icon there right after.

bennytrieu commented:
@LeviKuhaulua Changes listed below have been added onto my branch and is technically good to go for pull request review again however, I want to see if I can make my code a bit less repetitive. I want to give it till Friday to see if I can make it better. If not, whatever code is added up to "[Changed localStorage to sessionStorage](https://github.com/8bitUHM/fhsleoclub/pull/48/commits/d964ee94bcacf842dba67a8b28a7973950d101ad)" is good to go.

- Removed line of code in members.tsx that makes the roles lowercase
- Added code so that roles are lowercase before being added into the database
- Changed all the CRUD buttons to icon buttons
- Changed Add button to be on the next position that the newly added member/officer/club advisor could be in for all 3 sections

LeviKuhaulua commented:
@bennytrieu To reduce code duplication, it might be better to make an overall component for the members since the same HTML tags is being used 3 times and the same functions / logic is being applied 3 times as well.

bennytrieu commented:
- Cleaned up Update.tsx and AddMember.tsx a tiny bit
- Added 2 new files onto the lib folder
1. useAuthRedirect (code that redirects the user if they are not signed in)
2. useMemberValidation (Checks if a member's email already exists in the database)
- Moved duplicate HTML code onto another component called MemberData.tsx which stores:
1. handleDelete function
2. member's name, email, and role HTML
3. the update button and delete button
- Moved duplicate add member button HTML code onto another component called AddMemberButton.tsx

LeviKuhaulua reviewed (approved):
LGTM
